engine/crates/composition/src/subgraphs/keys.rs

```diff
@@ -121,15 +139,16 @@ impl NestedKeyFields {
 }
 
 /// Corresponds to an `@key` annotation.
-#[derive(Debug, PartialOrd, Ord, PartialEq, Eq)]
+#[derive(Debug, PartialOrd, PartialEq)]
 pub(crate) struct Key {
     selection_set: Vec<Selection>,
     resolvable: bool,
 }
 
-#[derive(PartialEq, Eq, PartialOrd, Ord, Debug)]
+#[derive(PartialEq, PartialOrd, Debug)]
 pub(crate) struct Selection {
     pub(crate) field: StringId,
+    pub(crate) arguments: Vec<(StringId, Value)>,
```

Comment:
Hum, unless I'm mistaken you're not validating the arguments, no?
IMHO composition should validate those and not bother with GraphQL coercion rules, it should just match.
Doesn't sound obvious to me that you'll be able to store it in a better format (`Vec<(InputValueDefinitionId, Value)>` typically) considering you're reading from SDL. But well

Though saying this but when providing a SDL schema in the CLI it'll bypass composition, so I'll have to validate in all cases. Errors at composition would be a lot friendlier for the user at least

Reply:
Yes I agree it's probably easier to validate at composition time, and worth the extra effort to put an InputValueDefinitionId there. I'll work on that.

Reply:
Done & pushed, let's see what the tests say.